aws/resource_aws_lex_bot.go

```diff
@@ -0,0 +1,484 @@
+}
+
+func expandLexPrompt(m map[string]interface{}) (prompt *lexmodelbuildingservice.Prompt) {
+	prompt = &lexmodelbuildingservice.Prompt{}
+	prompt.MaxAttempts = aws.Int64(int64(m["max_attempts"].(int)))
+	prompt.Messages = expandLexMessages(expandLexSet(m["message"].(*schema.Set)))
+
+	if v, ok := m["response_card"]; ok && v != "" {
+		prompt.ResponseCard = aws.String(v.(string))
+	}
+
+	return
+}
+
+func flattenLexIntents(intents []*lexmodelbuildingservice.Intent) (flattenedIntents []map[string]interface{}) {
+	for _, intent := range intents {
+		flattenedIntents = append(flattenedIntents, map[string]interface{}{
+			"intent_name":    aws.StringValue(intent.IntentName),
+			"intent_version": aws.StringValue(intent.IntentVersion),
+		})
+	}
+
+	return
+}
+
+// Expects a slice of maps representing the Lex objects.
+// The value passed into this function should have been run through the expandLexSet function.
+// Example: []map[intent_name: OrderFlowers intent_version: $LATEST]
+func expandLexIntents(rawValues []map[string]interface{}) (intents []*lexmodelbuildingservice.Intent) {
+	for _, rawValue := range rawValues {
+		intents = append(intents, &lexmodelbuildingservice.Intent{
+			IntentName:    aws.String(rawValue["intent_name"].(string)),
+			IntentVersion: aws.String(rawValue["intent_version"].(string)),
+		})
+	}
+
+	return
+}
+
+func resourceAwsLexBot() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsLexBotCreate,
+		Read:   resourceAwsLexBotRead,
+		Update: resourceAwsLexBotUpdate,
+		Delete: resourceAwsLexBotDelete,
+
+		Importer: &schema.ResourceImporter{
+			State: func(d *schema.ResourceData, m interface{}) ([]*schema.ResourceData, error) {
+				// The version is not required for import but it is required for the get request.
+				d.Set("version", "$LATEST")
+				return []*schema.ResourceData{d}, nil
+			},
+		},
+
+		Timeouts: &schema.ResourceTimeout{
+			Update: schema.DefaultTimeout(time.Minute),
+			Delete: schema.DefaultTimeout(5 * time.Minute),
+		},
+
+		Schema: map[string]*schema.Schema{
+			"abort_statement": {
+				Type:     schema.TypeList,
+				Required: true,
+				MinItems: 1,
+				MaxItems: 1,
+				Elem:     lexStatementResource,
+			},
+			"checksum": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"child_directed": {
+				Type:     schema.TypeBool,
+				Required: true,
+			},
+			"clarification_prompt": {
+				Type:     schema.TypeList,
+				Required: true,
+				MinItems: 1,
+				MaxItems: 1,
+				Elem:     lexPromptResource,
+			},
+			"description": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				Default:      "",
+				ValidateFunc: validation.StringLenBetween(0, 200),
+			},
+			"failure_reason": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"idle_session_ttl_in_seconds": {
+				Type:         schema.TypeInt,
+				Optional:     true,
+				Default:      300,
+				ValidateFunc: validation.IntBetween(60, 86400),
+			},
+			"intent": {
```

Comment:
Seeing as this is a list of of intents I would expect this argument to be called `intents`.

Reply:
It's a list but a list of objects not primitive types, so you end up with this